src/btree/multi.ts

```diff
@@ -0,0 +1,78 @@
+import { RangeResolver } from "../resolver";
+import { MemoryPointer } from "./node";
+
+export class LinkedMetaPage {
+	private resolver: RangeResolver;
+	private offset: number;
+
+	constructor(resolver: RangeResolver, offset: number) {
+		this.resolver = resolver;
+		this.offset = offset;
+	}
+
+	async root(): Promise<MemoryPointer | null> {
+		// we seek by 12 bytes since offset is 8 bytes, length is 4 bytes
+		const { data } = await this.resolver({
+			start: this.offset,
+			end: this.offset + 12,
+		});
```

Comment:
Since the reads are more expensive in js, it's reasonable to have a `getPage()` call, something like:

```
private async getPage(): Promise<ArrayBuffer> {
  return await this.resolver({
    start: this.offset,
    end: this.offset + 4096 - 1,
  });
}
```

Cache this promise in the `LinkedMetaPage` class.